framework/src/actions/SetupMeshAction.C

```diff
@@ -130,11 +130,9 @@ SetupMeshAction::SetupMeshAction(const InputParameters & params)
 void
 SetupMeshAction::setupMesh(MooseMesh * mesh)
 {
-  std::vector<BoundaryName> ghosted_boundaries =
-      getParam<std::vector<BoundaryName>>("ghosted_boundaries");
-
-  for (const auto & bnd_name : ghosted_boundaries)
-    mesh->addGhostedBoundary(mesh->getBoundaryID(bnd_name));
+  if (isParamValid("ghosted_boundaries"))
```

Comment:
same here and everywhere. The right fix is to set an empty vector as a default rather than adding the logic down here

Reply:
The reason to use isParamValid logic is to be more consistent with other default value check. For other data type, we allow empty values in addParam and catch them later.

Reply:
That's not a good reason IMO. You are implementing the behavior that is obtained by supplying an empty vector as a default. This adds logic which we already have implemented. And the way that we have it implemented has additional functionality, namely signaling to the documentation system and language server/json dump that the parameter has a default behavior. There is no advantage to using `isParamValid` her, only drawbacks.

Reply:
The default value in addParam for me is more like a suggested value for users, therefore it also shows in the documentation. A empty vector value showing there doesn't provide much useful information

Reply:
Sure, it shows that the code defaults to an empty list and not to some _unknown_ behavior that is hard coded in the specific object. I don't understand why this is even controversial for all other parameter types we supply a default value when we need one. This entire PR is about making things more consistent, so why are you using a different pattern to manually implement defaults just for std::vector parameters?

Reply:
These are two examples how other data type implemented(I can find more if needed). I don't know why vector parameter should all have empty default in addParam 

```
params.addParam<unsigned int>(
      "crack_front_point_index",
      "The index of the point on the crack front corresponding to this q function");
         _has_crack_front_point_index(isParamValid("crack_front_point_index")),

_crack_front_point_index(
    _has_crack_front_point_index ? getParam<unsigned int>("crack_front_point_index") : 0),
```

```
  params.addParam<std::string>("base_name", "Material property base name.");
  _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
```

Reply:
This is how it looks in documentation, not some _unknown_ behavior
<img width="768" alt="Screenshot 2023-08-29 at 2 56 57 PM" src="https://github.com/idaholab/moose/assets/118846840/ba96784a-5df4-49b2-a778-4f7acc44010a">

Reply:
Daniel is right here. It s better to have the default as it will show in the parameter as a default empty one. 
Here if base_name had a default you would see "" for the default or () or something, that would tell you more than what we have now (nothing)

Reply:
see here

Reply:
This is different from what Daniel said in my opinion.

Reply:
@dschwen has your comment been addressed

Reply:
yes